src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,421 @@
+            if (_protocolVersion < _firmata.QuerySupportedFirmataVersion())
+            {
+                throw new NotSupportedException($"Firmata version on board is {_protocolVersion}. Expected {_firmata.QuerySupportedFirmataVersion()}. They must be equal.");
+            }
+
+            Log($"Firmata version on board is {_protocolVersion}.");
+
+            _firmwareVersion = _firmata.QueryFirmwareVersion(out _firmwareName);
+
+            Log($"Firmware version on board is {_firmwareVersion}");
+
+            _firmata.QueryCapabilities();
+
+            _supportedPinConfigurations = _firmata.PinConfigurations; // Clone reference
+
+            Log("Device capabilities: ");
+            foreach (var pin in _supportedPinConfigurations)
+            {
+                Log(pin.ToString());
+            }
+
+            _firmata.EnableDigitalReporting();
+        }
+
+        /// <summary>
+        /// Firmware version on the device
+        /// </summary>
+        public Version FirmwareVersion
+        {
+            get
+            {
+                return _firmwareVersion ?? new Version();
+            }
+        }
+
+        /// <summary>
+        /// Name of the firmware.
+        /// </summary>
+        public string FirmwareName
+        {
+            get
+            {
+                return _firmwareName ?? string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Firmata version found on the board.
+        /// </summary>
+        public Version FirmataVersion
+        {
+            get
+            {
+                if (_protocolVersion == null)
+                {
+                    return new Version();
+                }
+
+                return _protocolVersion;
+            }
+        }
+
+        internal FirmataDevice Firmata
+        {
+            get
+            {
+                return _firmata ?? throw new ObjectDisposedException(nameof(ArduinoBoard));
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of capabilities per pin
+        /// </summary>
+        public ReadOnlyCollection<SupportedPinConfiguration> SupportedPinConfigurations
+        {
+            get
+            {
+                return _supportedPinConfigurations.AsReadOnly();
+            }
+        }
+
+        internal void Log(string message)
+        {
+            LogMessages?.Invoke(message, null);
+        }
+
+        private void FirmataOnError(string message, Exception? innerException)
+        {
+            LogMessages?.Invoke(message, innerException);
+        }
+
+        /// <summary>
+        /// Creates a GPIO Controller instance for the board. This allows working with digital input/output pins.
+        /// </summary>
+        /// <returns>An instance of GpioController, using an Arduino-Enabled driver</returns>
+        public GpioController CreateGpioController()
+        {
+            return CreateGpioController(DefaultPinNumberingScheme);
```

Comment:
I wouldn't create property DefaultPinNumberingScheme just so that you can have parameterless CreateGpioController

Reply:
This is in preparation of the behavior of the (future) board base class. There you'll define the DefaultPinNumberingScheme in the ctor, but not change it later.